schemas/specimen.json

```diff
@@ -0,0 +1,465 @@
+          "Lymphoid neoplasms"
+        ]
+      }
+    },
+    {
+      "name": "tumour_grade",
+      "description": " ",
+      "valueType": "string",
+      "restrictions": {
+        "required": true      }
+    },
+    {
+      "name": "tumour_staging_system",
+      "description": "Specify the tumour staging system used to assess the cancer at the time the tumour specimen was resected. Pathological classification is based on the clinical stage information (acquired before treatment) and supplemented/modified by operative findings and pathological evaluation of the resected specimen.",
+      "valueType": "string",
+      "restrictions": {
+        "required": true,
+        "codeList": [
+          "Binet",
+          "Rai",
+          "FIGO",
+          "Ann Arbor",
+          "Murphy",
+          "Lugano",
+          "AJCC 8th Edition",
+          "AJCC 7th Edition",
+          "AJCC 6th Edition",
+          "AJCC 5th Edition",
+          "AJCC 4th Edition",
+          "AJCC 3rd Edition",
+          "AJCC 2nd Edition",
+          "AJCC 1st Edition"
+        ]
+      }
+    },
+    {
+      "name": "pathological_t_category",
+      "description": "The code to represent the stage of cancer defined by the size or contiguous extension of the primary tumour (T), according to criteria based on multiple editions of the AJCC's Cancer Staging Manual.",
+      "valueType": "string",
+      "restrictions": {
+        "script": "validateWithMagic(only needed if tumour_staging_system=AJCC)"
+      }
+    },
+    {
+      "name": "pathological_n_category",
+      "description": "The code to represent the stage of cancer defined by whether or not the cancer has reached nearby lymph nodes (N), according to criteria based on multiple editions of the AJCC's Cancer Staging Manual.",
+      "valueType": "string",
+      "restrictions": {
+        "script": "validateWithMagic(only needed if tumour_staging_system=AJCC)"
+      }
+    },
+    {
+      "name": "pathological_n_category",
+      "description": "The code to represent the stage of cancer defined by whether there are distant metastases (M), meaning spread of cancer to other parts of the body, according to criteria based on multiple editions of the AJCC's Cancer Staging Manual.",
+      "valueType": "string",
+      "restrictions": {
+        "script": "validateWithMagic(only needed if tumour_staging_system=AJCC)"
+      }
+    },
+    {
+      "name": "percent_tumour_cells",
+      "description": "Indicate a value, in decimals, that represents the percentage of infiltration by tumour cells in a specimen.",
+      "valueType": "number",
+      "restrictions": {
+        "required": true
+      }
+    },
+    {
+      "name": "percent_proliferating_cells",
+      "description": "Indicate a value, in decimals, that represents the count of proliferating cells determined during pathologic review of the specimen.",
+      "valueType": "number",
+      "restrictions": {
+        "required": true
```

Comment:
is it possible to restriction range? ie, `>=0` and `<=1`

Reply:
we can do that via scripts

Reply:
asking to see if there is an explicit restriction for that

Reply:
https://github.com/overture-stack/lectern/issues/43 <- have requested addition in lectern